synfig-studio/src/gui/states/state_bone.cpp

```diff
@@ -1282,3 +1338,47 @@ StateBone_Context::_on_signal_value_desc_set(ValueDesc value_desc,ValueBase valu
 		}
 	}
 }
+
+void
+StateBone_Context::toggle_layer_skel()
+{
+	if(!layer_skel_flag)
+	{
+		set_layer_skel_flag(true);
+		set_layer_skel_deform_flag(false);
+
+		layer_skel_flag = true;
+		layer_skel_deform_flag = false;
+
+		save_settings();
+		c_layer = 0;
+		update_tool_options(0);
+		load_settings();
+	}
+
+	if(get_layer_skel_flag() +
+	   get_layer_skel_deform_flag() == 0)
+		set_layer_skel_flag(true);
+}
```

Comment:
I think instead of declaring new variables  `layer_skel_flag` and `layer_skel_deform_flag` and their setters and getters, `toggle_button.set_active()` and `toggle_button.get_active()` can do the trick since we're only dealing with boolean values.

Something like this:
```c++
void
StateBone_Context::toggle_layer_skel()
{
	if(layer_skel_togglebutton.get_active())
	{
		layer_skel_deform_togglebutton.set_active(false);

		save_settings();
		c_layer = 0;
		update_tool_options(0);
		load_settings();
	}

	if(layer_skel_togglebutton.get_active() +
	   layer_skel_deform_togglebutton.get_active() == 0)
		layer_skel_togglebutton.set_active(true);
}
```

Reply:
I don't remember why I choose to keep flags as used currently for other tools, but looking at the code as of now as you suggested I don't think they're necessary... Checking now, will report ASAP.

Reply:
@Keyikedalube I've removed them, works fine with less stuff :wink: